src/annotation.js

```diff
@@ -26,6 +26,19 @@ export default {
 
 	beforeUpdate(chart, args, options) {
 		const annotationOptions = options || args;
+
+		const array = new Array();
+		if (!!annotationOptions.annotations) {
+			Object.keys(annotationOptions.annotations).forEach(key => {
+				let value = annotationOptions.annotations[key];
+				if (isObject(value)) {
+					value.id = key;
+					array.push(value);
+				}
+			});
+		}
+		annotationOptions.annotations = array;
```

Comment:
Still internally an array, so we could as well support both ways?
What is the ID used for? (I did not need it in my use cases)

The array is easier, when you need to dynamically add new annotations (like the updates example).

Reply:
> Still internally an array, so we could as well support both ways?

If you use the array (both defaults and chart), the config is not merged, you have only the chart, if I'm not wrong.
If you use the array on defaults and a object in a chart (or viceversa), you have only the chart settings.
Sounds not doable.

> The array is easier, when you need to dynamically add new annotations (like the updates example).

Yes, you're right, it's easier. Nevertheless using a counter you can do it (done on the updates sample) but it's not possible to use the defaults with arrays, not doable. 
I don't know what's better. In my use cases, I didn't have any need to add an annotation at runtime and if needed, I'd create an id. But, that's only my personal opinion.

Reply:
Anyway, let me check. Because I think we can implement both ways, with limitation that the defaults are not merged.

Reply:
Now you can manage the annotations both by an Array (as it was before) and by an Object.
The default is an object instead of an array. ok?.